packages/core/lib/console.js

```diff
@@ -232,9 +264,38 @@ class Console extends EventEmitter {
     const contextVars = {};
 
     abstractions.forEach(abstraction => {
-      contextVars[abstraction.contract_name] = abstraction;
+      // don't overwrite Node's native objects - only load contracts
+      // into the repl context when no conflict exists
+      if (
+        this.replContextNameConflicts[abstraction.contract_name].conflict ===
+        false
+      ) {
+        contextVars[abstraction.contract_name] = abstraction;
+      }
     });
 
+    if (this.warnUserOfConflicts) {
+      let contractNames = [];
+      for (const name in this.replContextNameConflicts) {
+        if (
+          this.replContextNameConflicts[name].conflict === true &&
+          this.replContextNameConflicts[name].warned === false
+        ) {
+          this.replContextNameConflicts[name].warned = true;
+          contractNames.push(name);
+        }
+      }
```

Comment:
This piece of logic reconstructs the information `recordNameConflicts` already knew, as well as update the repl's known collisions.